docs: ✏️ review config guide (#257)
seedcase-project/seedcase-flower: docs/guide/config.qmd

---
title: "Configuration files"

> description: "How to configure Flower's settings and output options using a configuration file."

Author: ```suggestion
description: "Configure Flower's settings and output options using a configuration file."
```

Author: To match the description of the other guides.

order: 2
---

Flower can be configured via the command line interface (CLI), the
Python interface, or a configuration file. This guide covers
configuration files---for the CLI, see the
[CLI guide](/docs/guide/cli.qmd). We don't have a guide for the Python
interface, as Flower is primarily designed to be used as a CLI tool. If
you're interested in the Python interface, check out the
[Python](/docs/design/interface/python.qmd) design document.

## Configuration files

A configuration file lets you set your preferences once, so you don't
have to repeat them every time you run `flower build`. Flower supports

> two configuration files: `.flower.toml` and `pyproject.toml`. Both

Reviewer: ```suggestion
two configuration files: `.flower.toml` and `pyproject.toml`. These files
```

"Both" I interpret as the user needs to use both.

Author: Ah, I see. Good catch, thanks!

> should be placed in the root of your project — Flower will find them

Reviewer: ```suggestion
should be placed in the root of your project---Flower will find them
```

Author: Those damn em-dashes 👀

automatically.

Use `.flower.toml` if you prefer a dedicated file for Flower's settings,

> or `pyproject.toml` if you already use it for other Python tools and

Reviewer: ```suggestion
or `pyproject.toml` if your Data Package is also a Python project and you
use the `pyproject.toml` to store other config settings, and you
```

prefer to keep everything in one place.

::: callout-note
Settings provided via the CLI or Python interface always take priority

> over the configuration file. Some settings — like specifying a remote

Reviewer: ```suggestion
over the configuration file. Some settings---like specifying a remote
```

Your computer seems to convert them to smart dashes, can you turn that off?

Author: It's me who forgets to use em dashes 🤖

> source or configuring the `view` command — can only be set via the CLI

Reviewer: ```suggestion
source or configuring the `view` command---can only be set via the CLI
```
